server/container/util/src/main/java/org/apache/james/util/ReactorUtils.java

```diff
@@ -18,10 +18,66 @@
  ****************************************************************/
 package org.apache.james.util;
 
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Spliterator;
+import java.util.stream.Stream;
+
+import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 public class ReactorUtils {
     public static <T> Mono<T> executeAndEmpty(Runnable runnable) {
         return Mono.fromRunnable(runnable).then(Mono.empty());
     }
+
+    public static InputStream toInputStream(Flux<byte[]> byteArrays) {
+        return new StreamInputStream(byteArrays.toStream());
+    }
+
+    private static  class StreamInputStream extends InputStream {
+        private final Stream<byte[]> source;
+
+        private Spliterator<byte[]> spliterator;
+        private ByteArrayInputStream currentItemByteStream;
+
+        StreamInputStream(Stream<byte[]> source) {
+            this.source = source;
+            this.currentItemByteStream = null;
+        }
+
+        @Override
+        public int read() throws IOException {
+            if (spliterator == null) {
+                spliterator = source.spliterator();
```

Comment:
You can do this in the constructor and not having the check in each call to read.